### PR TITLE
Add filter contact list based on salary feature

### LINKED ISSUE #64: As a Computer Science job seeker, I can filter my interview contact list based on their salary ranges
so that I can look at relevant interviews more conveniently based on the salary range

### PR DESCRIPTION
**Pull request by Lalelulilulela**
closes https://github.com/AY2324S2-CS2103T-T08-3/tp/issues/64
- Command Format: filter s/SALARY_RANGE [MORE SALARY_RANGE]...
  - e.g. find s/2000 5000-8000 >=10000 would return any contact with salary range 2000 or salary range containing any value within 5000-8000 or any salary more than or equals to 10000

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+public class FilterCommandParser implements Parser<FilterCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FilterCommand
+     * and returns a FilterCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FilterCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(
```
Comment (dabzpengu) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java:25]: nice use of the tokenizer!

File: src/main/java/seedu/address/model/person/SalaryRange.java
Comment (dabzpengu) [inline src/main/java/seedu/address/model/person/SalaryRange.java:1]: the oop here is a bit too low-level cause you created a whole separate class for salary range, maybe try to modify salary class to work with your parser? if i'm not mistaken, the current Salary class is already able to take range values

**Review by dabzpengu — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/person/SalaryRange.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            return isValidRange(test);
+        } else if (test.contains(">=")) {
+            return isValidMin(test);
+        } else if (test.contains("<=")) {
+            return isValidMax(test);
+        } else {
+            return isValidNum(test);
+        }
+    }
+    private static boolean isValidRange(String test) {
```
Comment (ashleyy2444) [inline src/main/java/seedu/address/model/person/SalaryRange.java]: Can try to ensure that all methods are being tested for to increase code coverage

File: src/main/java/seedu/address/model/person/SalaryContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class SalaryContainsKeywordsPredicate implements Predicate<Person> {
```
Comment (ashleyy2444) [inline src/main/java/seedu/address/model/person/SalaryContainsKeywordsPredicate.java:12]: Good and intuitive naming of class

File: src/main/java/seedu/address/model/person/SalaryRange.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        checkArgument(isValidSalaryRange(salaryRange), MESSAGE_CONSTRAINTS);
+        parseSalaryRange(salaryRange);
+    }
+
+    /**
+     * Constructs a {@code Salary}.
+     *
+     * @param salaryRange A valid salary in string format.
+     */
+    public void parseSalaryRange(String salaryRange) {
```
Comment (ashleyy2444) [inline src/main/java/seedu/address/model/person/SalaryRange.java:46]: There is some code duplication in the way different cases are handled. Might want to consider refactoring to reduce redundancy.

**Comment by ashleyy2444** [comment]
LGTM

**Pull request merged**